include/eeprom/core/accessor.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "addresses.hpp"
+#include "task.hpp"
+#include "types.hpp"
+
+namespace eeprom {
+namespace accessor {
+
+/**
+ * Interface of an object that listens for reads
+ */
+template <typename T>
+class ReadListener {
+  public:
+    ReadListener() = default;
+    ReadListener(const ReadListener&) = default;
+    auto operator=(const ReadListener&) -> ReadListener& = default;
+    ReadListener(ReadListener&&) noexcept = default;
+    auto operator=(ReadListener&&) noexcept -> ReadListener& = default;
+
+    virtual ~ReadListener() = default;
+    virtual void on_read(const T&) = 0;
+};
+
+template <task::TaskClient EEPromTaskClient, typename T,
```

Comment:
what's `T` here? can we restrict it more tightly with a concept or a comment or something